docs/pages/tv-settings_halSpec.md

```diff
@@ -0,0 +1,514 @@
+    HAL-->>Caller:return
+    Caller->>HAL: SetTVPictureMode()
+    Note over HAL: Set the default picture mode entertainment
+    HAL->>OPPDB: Read associated picture properties
+    OPPDB-->>HAL:return
+    Note over HAL: Reload gamma and white balance if there is a colour temperature value change.
+    HAL->>OPPDB: Read associated Gamma and White balance
+    OPPDB-->>HAL:return
+    Note over HAL: Reload TMAX if there is a LDIM level change
+    HAL->>OPPDB: Read associated TMAX value
+    OPPDB-->>HAL:return
+    Note over HAL: Apply the new picture properties, gamma, TMAX and white balance if they have changed.
+    HAL->>Driver: Apply new picture properties
+    Driver-->>HAL:return
+    HAL->>OPPDB: Update the Picture mode association table
+    Note over HAL: Associate new picture mode to current video format and current video source
+    OPPDB-->>HAL:return
+    HAL-->>Caller:return
+
+    Caller->>HAL: RegisterCallBack
+    Note over HAL:RegisterCallBack for Format/Resolution/FrameRate/VideoContent Change
+    HAL-->>Caller:return
+
+    Driver-->>HAL:Notify on video format/framerate/resolution/videocontent/videosource change
+    Note over HAL: Reload associated Picture mode and all associated picture properties if there is video format and video source change.
+    Note over HAL: However video source change will not be notified to caller
+    HAL->>OPPDB: Read associated picture mode
+    OPPDB-->>HAL:return
+    HAL->>OPPDB: Read associated picture properties
+    OPPDB-->>HAL:return
+    Note over HAL: Reload gamma and white balance if there is a colour temperature value change.
+    HAL->>OPPDB: Read associated Gamma and White balance
+    OPPDB-->>HAL:return
+    Note over HAL: Reload TMAX if there is a LDIM level change
+    HAL->>OPPDB: Read associated TMAX value
+    OPPDB-->>HAL:return
+    Note over HAL: Apply the new picture properties, gamma and white balance if they have changed.
+    HAL->>Driver: Apply new picture properties
+    Driver-->>HAL: return
+    HAL-->>Caller:Notify on video format/framerate/resolution/content change
+```
+
+##### Set/Get/Save Picture Quality Parameter Sequence
+```mermaid
+sequenceDiagram
+participant Caller as Caller
+    participant HAL as TV Settings HAL
+    participant OPPDB as Override Picture Profile DB
+    participant DPPDB as Default Picture Profile DB
+    participant Driver as SoC
+
+    Caller->>HAL: tvSettings_SetMethods
+    Note over HAL: APIs to set the Picture Quality Parameters
+    HAL->>Driver:Sets the PQ Parameters
+    Driver-->>HAL:return
+    HAL->>OPPDB: Saves the new PQ Parameter
+    OPPDB-->>HAL:return
+    HAL-->>Caller:return
+
+    Caller->>HAL: tvSettings_GetMethods
+    Note over HAL: APIs to get the PQ Parameters
+    HAL->>Driver:Gets the PQ Parameters
+    Driver-->>HAL:return
+    HAL-->>Caller:return
+
+    Caller->>HAL: tvSettings_SaveMethods
+    Note over HAL: APIs to save the Picture Quality Parameters
+    HAL->>OPPDB:Save the PQ Parameters

```

Comment:
Can we define which Get API will get from SoC, which ones will get from OPPDB and which one will get from DPPDB

Reply:
This one not addressed?